### PR TITLE
feat: allow selection of individual service in Trace filter page

### LINKED ISSUE #2470: Allow selection of individual service in Trace filter page
Currently, if a user wants to select a service he must click on Clear All and then choose his service. This affects the performance of the page as well as friction for common action by users.

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #2470 

## Demo
https://www.loom.com/share/ada8b2fa84284c87babbbccb191d6ba5

### CODE REVIEW

**Comment by gitstart** [comment]
Hi @palashgdev, please review the PR 🙏

**Comment by palashgdev** [comment]
@makeavish can you review this PR

**Comment by palashgdev** [comment]
but i feel that code will not be able to solve not sure about this one but let @makeavish confirm this one first

**Review by makeavish — APPROVED** [review]

File: frontend/src/container/Trace/Filters/Panel/PanelBody/Common/Checkbox.tsx [hunk 1]
```diff
@@ -35,22 +35,29 @@ function CheckBoxComponent(props: CheckBoxProps): JSX.Element {
 	const dispatch = useDispatch<Dispatch<AppActions>>();
 
 	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [timeoutId, setTimeoutId] = useState<NodeJS.Timeout>();
```
Comment (YounixM) [inline frontend/src/container/Trace/Filters/Panel/PanelBody/Common/Checkbox.tsx:38]: We should not use NodeJS.Timeout

File: frontend/src/container/Trace/Filters/Panel/PanelBody/Common/Checkbox.tsx [hunk 2]
```diff
@@ -159,6 +170,19 @@ function CheckBoxComponent(props: CheckBoxProps): JSX.Element {
 		}
 	};
 
+	const onCheckHandler: React.MouseEventHandler<HTMLElement> = async (
+		e,
+	): Promise<void> => {
+		if (e.detail === 1) {
+			const id = setTimeout(updateFilter, 500);
+			setTimeoutId(id);
```
Comment (YounixM) [inline frontend/src/container/Trace/Filters/Panel/PanelBody/Common/Checkbox.tsx:178]: Why do we need the timeout ?

**Comment by makeavish** [comment]
This has improved with new explorer page

**Pull request closed without merging**